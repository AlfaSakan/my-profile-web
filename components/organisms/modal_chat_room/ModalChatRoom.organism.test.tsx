import renderer from 'react-test-renderer';
import ModalChatRoom from './ModalChatRoom.organism';

describe('HeaderBody component test', () => {
  it('should render correctly', () => {
    const onChangeName = jest.fn();
    const onChangeDescription = jest.fn();
    const onClickSubmit = jest.fn();
    const onClose = jest.fn();

    const tree = renderer
      .create(
        <ModalChatRoom
          descVal="description"
          nameVal="name value"
          onChangeName={onChangeName}
          onChangeDescription={onChangeDescription}
          isOpen
          onClickSubmit={onClickSubmit}
          onClose={onClose}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
