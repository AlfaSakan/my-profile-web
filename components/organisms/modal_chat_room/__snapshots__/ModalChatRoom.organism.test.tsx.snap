// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBody component test should render correctly 1`] = `
<div
  className="fixed h-full w-full bg-black-200 top-0 flex items-center justify-center"
  onClick={[MockFunction]}
>
  <div
    className="bg-primary flex flex-col md:w-1/2 md:mx-0 h-1/2 rounded-2xl w-full mx-10"
    onClick={[Function]}
  >
    <div
      className="flex items-center justify-between px-8 py-4 bg-secondary rounded-t-2xl"
    >
      <p
        className="text-sm"
      >
        Buat Chat Room Baru
      </p>
      <div
        className="cursor-pointer"
        onClick={[MockFunction]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </div>
    </div>
    <div
      className="flex flex-1 flex-col px-8 py-4 border-y justify-between"
    >
      <div
        className="flex-col"
      >
        <p>
          Nama Chat Room
        </p>
        <input
          className="w-full focus:outline-none rounded-md py-2 px-4 border py-3 mt-2"
          onChange={[Function]}
          placeholder="Type a message"
          type="text"
          value="name value"
        />
      </div>
      <div
        className="flex-col"
      >
        <p>
          Deskripsi
        </p>
        <input
          className="w-full focus:outline-none rounded-md py-2 px-4 border py-3 mt-2"
          onChange={[Function]}
          placeholder="Type a message"
          type="text"
          value="description"
        />
      </div>
      <div
        className="flex flex-col relative"
      >
        <p
          className="text-sm"
        >
          Anggota
        </p>
        <div
          className="border py-3 px-4 bg-white rounded-md flex gap-2 items-center cursor-pointer h-12 mt-2"
          onClick={[Function]}
        />
      </div>
    </div>
    <div
      className="flex items-center justify-end px-8 py-4"
    >
      <button
        className="flex items-center justify-center w-32 h-12 rounded-lg bg-secondary"
        onClick={[MockFunction]}
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      >
        <p
          className="text-sm"
        >
          Login
        </p>
      </button>
    </div>
  </div>
</div>
`;
